### ISSUE TITLE
Children of disabled blocks showing with useMemoized

### ISSUE THREAD

**Issue by leigeber** [spicywebau/craft-neo#282]
#### Description

When using the useMemoized method on blocks prior to looping, children of disabled top-level blocks are still being displayed. Without useMemoized things work as expected.

#### Steps to reproduce
1. Look through blocks and call ```block.useMemoized(blocks)```
2. In the loop ```block in blocks if block.level == 1``` see that the children of the disabled blocks flow into the previous enabled top-level block

#### Other information
- Neo version: 2.5.8
- Craft version: Craft Pro 3.3.15
- (if multi-site) Are the affected Neo fields set to manage blocks on a per-site basis:
- (if multi-site, > 2.4.0) What is the propagation method of the neo field:
- Is eager-loading used? Yes

**Comment by pvldigital** [comment]
Hi @leigeber,

I have been testing but can't replicate it. Is it possible for you to extract and send your template code to plugins@spicyweb.com.au?

**Comment by ttempleton** [comment]
Closing this issue, due to inactivity and not being reproducible. If the issue persists for you with the latest Craft and Neo releases, please let us know and we will reopen the issue.

**Comment by stevendierick** [comment]
I'm having the same issue. I'm trying to eager load my Neo field, which has one top level block with several children. The children still show if I disable a top level block. It's as if the children get appended to the previous top level block.

Works perfectly fine if I remove the eager loading bit of code.

**content-builder.twig:**

``` twig
{% set contentBuilder = entry.contentBuilder.all() %}

{% for block in contentBuilder %}
    {% do block.useMemoized(contentBuilder) %}
{% endfor %}

{% for block in contentBuilder %}
    {% if block.type.handle == 'topLevel' %}
        {% include '_content-builder/top-level' ignore missing %}
    {% endif %}
{% endfor %}
```

**top-level.twig:**

``` twig
<section class="section ...">
    {% for block in section.children.all() %}
        {% switch block.type.handle %}
            {% case 'text' %}
                {% include '_content-builder/text' ignore missing %}
            {% case 'button' %}
                {% include '_content-builder/button' ignore missing %}
            {% case 'images' %}
                {% include '_content-builder/images' ignore missing %}
        {% endswitch %}
    {% endfor %}
</section>
```

Neo version: 2.8.17
Craft CMS version: 3.5.18

**Comment by ttempleton** [comment]
Thanks for the template example @stevendierick, with that I was able to reproduce the bug, and it should now be fixed in Neo 2.8.18.
